rust/wal3/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+#![doc = include_str!("../README.md")]
+
+use serde::{Deserialize, Serialize};
+
+mod backoff;
+
+use backoff::ExponentialBackoff;
+
+/////////////////////////////////////////////// Error //////////////////////////////////////////////
+
+/// Error captures the different error conditions that can occur when interacting with the log.
+#[derive(Clone, Debug, Default)]
```

Comment:
Can we use thiserror here? Its what we use across the codebase to add error messages to an enum